File: src/mover.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"

#include <sstream>

int main(int argc, char **argv)
{

  ros::init(argc, argv, "mover");

  ros::NodeHandle n;

  ros::Publisher movement_pub = n.advertise<geometry_msgs::Twist>("cmd_vel", 1);

  // We can define a custom rate for our messages here (in Hz)
  ros::Rate loop_rate(10);

  while (ros::ok())
  {
    geometry_msgs::Twist msg;

    float freq = 2.0;

    ros::Time ctime = ros::Time::now();

    msg.linear.x = sin(2*M_PI*freq*ctime.toSec());

    movement_pub.publish(msg);

    ros::spinOnce();

    loop_rate.sleep();
  }


  return 0;
}

File: scripts/mover.py
#! /usr/bin/env python
import rospy
import math
from geometry_msgs.msg import Twist

def mover():
    pub = rospy.Publisher('cmd_vel', Twist, queue_size=1)
    rospy.init_node('talker', anonymous=True)
    rate = rospy.Rate(10)
    vel = Twist()
    freq = 0.2
    while not rospy.is_shutdown():
        vel.linear.x = math.sin(2*math.pi*freq*rospy.get_time())
        pub.publish(vel)
        rate.sleep()

if __name__ == '__main__':
    try:
        mover()
    except rospy.ROSInterruptException:
        pass
